lib/retry.js

```diff
@@ -95,13 +99,13 @@ export default function retry(times, task, callback) {
         throw new Error("Invalid arguments for async.retry");
     }
 
-
     var attempts = [];
-    while (opts.times) {
-        var isFinalAttempt = !(opts.times -= 1);
+    for (var i = 1; i < opts.times + 1; i++) {
+        var isFinalAttempt = (i == opts.times);
```

Comment:
I needed to pass the retryCount to the interval function but the loop was counting backwards and destroying opts.times.